packages/graphql-generator/src/typescript.ts

```diff
@@ -0,0 +1,40 @@
+export type ModelsTarget = 'java' | 'swift' | 'javascript' | 'typescript' | 'dart' | 'introspection';
+export type StatementsTarget = 'javascript' | 'graphql' | 'flow' | 'typescript' | 'angular';
+export type TypesTarget = 'json' | 'swift' | 'typescript' | 'flow' | 'scala' | 'flow-modern' | 'angular';
+
+export type FileExtension = 'js' | 'graphql' | 'ts';
+
+export type GenerateTypesOptions = {
+  schema: string;
+  target: TypesTarget;
+  queries?: string[];
+  introspection?: boolean;
+  only?: string; // only used when target is swift
```

Comment:
I actually hadn't fully investigated this yet (I was going to take a look later). But ends up being something we can probably remove.

https://github.com/aws-amplify/amplify-codegen/blob/e8d63f07dd216edc2ea0a418927302390c18634c/packages/graphql-types-generator/src/cli.js#L47-L51

> Parse all input files, but only output generated code for the specified file [Swift only]

So this could probably be removed from the `graphql-generator` level and remain a concern of `amplify-codegen`. I'll fix this in a later PR.